chemprop/cli/train.py

```diff
@@ -651,6 +652,11 @@ def build_model(args, train_dset: MolGraphDataset | MulticomponentDataset) -> MP
         )
     else:
         criterion = None
+    if args.metrics is not None:
+        # TODO: AUROCMetric takes `task` as an argument, but we don't expose that through the command line. Should we?
```

Comment:
The `task` in AUROCMetric is a helper argument required by the `F.auroc` function. The available options are `["binary", "multiclass", "multilabel"]`, so we should be able to set them based on the predictor type.

```
@PredictorRegistry.register("classification")
class BinaryClassificationFFN(BinaryClassificationFFNBase):
    n_targets = 1
    _default_criterion = BCELoss()
    _default_metric = AUROCMetric(task="binary")

    def forward(self, Z: Tensor) -> Tensor:
        Y = super().forward(Z)

        return Y.sigmoid()

    def train_step(self, Z: Tensor) -> Tensor:
        return super().forward(Z)


@PredictorRegistry.register("classification-dirichlet")
class BinaryDirichletFFN(BinaryClassificationFFNBase):
    n_targets = 2
    _default_criterion = BinaryDirichletLoss()
    _default_metric = AUROCMetric(task="binary")

    def forward(self, Z: Tensor) -> Tensor:
        Y = super().forward(Z)
        alpha, beta = torch.chunk(Y, 2, 1)

        return beta / (alpha + beta)

    def train_step(self, Z: Tensor) -> Tensor:
        Y = super().forward(Z)

        F.softplus(Y) + 1
```

Reply:
I will remove the comment and `task=None`. Either way this will fail if the user tries to specify `AUROCMetric` from the command line, but they can access it by specifying no metrics. It is the default for some predictors as you said. 

The limitation of this is that users can't use `AUROCMetric` and other metrics at the same time. I don't know how likely that use case is, but if it is we can address it then. I think this is better than adding a CLI argument for a niche use case.